drivers/overlay/overlay.go

```diff
@@ -1205,17 +1205,17 @@ func (d *Driver) getAllImageStores() []string {
 	return additionalImageStores
 }
 
-func (d *Driver) dir2(id string, useImageStore bool) (string, string, bool) {
-	var homedir string
-
+func (d *Driver) homeDir(useImageStore bool) string {
```

Comment:
I’m confused by the terminology. We have `d.homeDir` and `d.home`, and those are different values with different semantics.

What do the values actually _mean_?

(I mean, I’m generally confused and scared by these code paths and options and indirections. This is just a very clear manifestation of my general confusion. But, still.)

---

`pruneStagingDirectories` hard-codes `filepath.Join(d.home, stagingDir)`, not using the `useImageStore` condition. Which of the two is correct?

Reply:
thanks, I've renamed the function, hopefully it is clearer now.

`pruneStagingDirectories` was using the wrong location, I've fixed this too

Reply:
I am still generally confused ;), but making the names are much more distinct does help.